shell/platform/linux/fl_settings_portal.cc

```diff
@@ -0,0 +1,263 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/fl_settings_portal.h"
+
+#include <gio/gio.h>
+#include <glib.h>
+
+static constexpr char kPortalName[] = "org.freedesktop.portal.Desktop";
+static constexpr char kPortalPath[] = "/org/freedesktop/portal/desktop";
+static constexpr char pPortalSettings[] = "org.freedesktop.portal.Settings";
+
+struct FlSetting {
+  const gchar* ns;
+  const gchar* key;
+  const GVariantType* type;
+};
+
+static constexpr char kXdgAppearance[] = "org.freedesktop.appearance";
+static const FlSetting kColorScheme = {
+    kXdgAppearance,
+    "color-scheme",
+    G_VARIANT_TYPE_UINT32,
+};
+
+static constexpr char kGnomeDesktopInterface[] = "org.gnome.desktop.interface";
+static const FlSetting kClockFormat = {
+    kGnomeDesktopInterface,
+    "clock-format",
+    G_VARIANT_TYPE_STRING,
+};
+static const FlSetting kGtkTheme = {
+    kGnomeDesktopInterface,
+    "gtk-theme",
+    G_VARIANT_TYPE_STRING,
+};
+static const FlSetting kTextScalingFactor = {
+    kGnomeDesktopInterface,
+    "text-scaling-factor",
+    G_VARIANT_TYPE_DOUBLE,
+};
+
+static const FlSetting all_settings[] = {
+    kClockFormat,
+    kColorScheme,
+    kGtkTheme,
+    kTextScalingFactor,
+};
+
+static constexpr char kClockFormat12Hour[] = "12h";
+static constexpr char kGtkThemeDarkSuffix[] = "-dark";
+
+typedef enum { DEFAULT, PREFER_DARK, PREFER_LIGHT } ColorScheme;
+
+struct _FlSettingsPortal {
+  GObject parent_instance;
+
+  GDBusProxy* dbus_proxy;
+  GVariantDict* values;
+};
+
+static void fl_settings_portal_iface_init(FlSettingsInterface* iface);
+
+G_DEFINE_TYPE_WITH_CODE(FlSettingsPortal,
+                        fl_settings_portal,
+                        G_TYPE_OBJECT,
+                        G_IMPLEMENT_INTERFACE(fl_settings_get_type(),
+                                              fl_settings_portal_iface_init))
+
+#define FL_SETTINGS_PORTAL_GET_PRIVATE(portal) \
+  ((FlSettingsPortalPrivate*)fl_settings_portal_get_instance_private(portal))
+
+static const gchar* fl_settings_portal_format_key(const FlSetting* setting) {
```

Comment:
Having the `fl_settings_portal_` prefix makes this look like a method on FlSettingsPortal. I suggest renaming it to `format_key` (and the same for other static functions in this file).